File: java/app-encryption/src/test/java/com/godaddy/asherah/crypto/engine/bouncycastle/BouncyAes256GcmCryptoTest.java
package com.godaddy.asherah.crypto.engine.bouncycastle;

import com.godaddy.asherah.crypto.engine.GenericAeadCryptoTest;
import com.godaddy.asherah.crypto.envelope.AeadEnvelopeCrypto;

class BouncyAes256GcmCryptoTest extends GenericAeadCryptoTest {
  @Override
  protected AeadEnvelopeCrypto getCryptoInstance() {
    return new BouncyAes256GcmCrypto();
  }
}

File: csharp/AppEncryption/AppEncryption.Tests/Crypto/Engine/BouncyCastle/BouncyAes256GcmCryptoTest.cs
using GoDaddy.Asherah.Crypto.Engine.BouncyCastle;
using GoDaddy.Asherah.Crypto.Envelope;
using Microsoft.Extensions.Configuration;

namespace GoDaddy.Asherah.AppEncryption.Tests.Crypto.Engine.BouncyCastle
{
    public class BouncyAes256GcmCryptoTest : GenericAeadCryptoTest
    {
        public BouncyAes256GcmCryptoTest(ConfigFixture configFixture)
            : base(configFixture)
        {
        }

        protected override AeadEnvelopeCrypto GetCryptoInstance(IConfiguration configuration)
        {
            return new BouncyAes256GcmCrypto(configuration);
        }
    }
}
